crates/swc_plugin_runner/src/lib.rs

```diff
@@ -143,14 +148,39 @@ pub fn apply_js_plugin(
     (|| -> Result<_, Error> {
         let instance = load_plugin(path, cache)?;
 
-        let plugin_process = instance
+        let plugin_process_wasm_exported_fn = instance
             .exports
             .get_native_function::<(i32, u32), (i32, i32)>("process")?;
 
         let (alloc_ptr, len) = copy_memory_to_instance(&instance, &program)?;
-        let (_returned_ptr, _returned_len) = plugin_process.call(alloc_ptr, len)?;
+        let (returned_ptr, returned_len) = plugin_process_wasm_exported_fn.call(alloc_ptr, len)?;
+        let returned_len = returned_len.try_into().unwrap();
 
-        Ok(program)
+        // Reconstruct AlignedVec from ptr returned by plugin
+        let memory = instance.exports.get_memory("memory")?;
+        let ptr: WasmPtr<u8, Array> = WasmPtr::new(returned_ptr as _);
+
+        let mut transformed_serialized = AlignedVec::with_capacity(returned_len);
+
+        // Deref & read through plguin's wasm memory space via returned ptr
+        let derefed_ptr = ptr
+            .deref(memory, 0, returned_len.try_into().unwrap())
+            .unwrap();
```

Comment:
these are just good enough verifies end-to-end workflow, further polishing should introduce better handlings (refer //todo comment).